### PR TITLE
allow connecting to custom server

### PR DESCRIPTION
**Pull request by saadataz**
To facilitate development, allow specifying server URL in basic example

### CODE REVIEW

**Comment by kateract** [comment]
Not sure how this facilitates development, rather encourages incorrect behavior I would think.  This capability was specifically removed a few commits ago in favor of an automatic (and preferred) method.

File: examples/basic.ts [hunk 1]
```diff
@@ -72,6 +72,7 @@ function makeControls(amount: number): IControlData[] {
 client.open({
     authToken: process.argv[2],
     versionId: parseInt(process.argv[3], 10),
+    url: process.argv[4] || null,
```
Comment (connor4312) [inline examples/basic.ts]: I'd rather not push 'weirdness' to a public example for functionality that's only useful for internal testing.

**Comment by ProbablePrime** [comment]
Ahh we nuked this when i added endpoint discovery. Yeah it shouldn't be in the examples in our master branch

File: src/GameClient.ts [hunk 2]
```diff
@@ -34,7 +34,7 @@ export class GameClient extends Client {
         .then(endpoints => {
             return super.open({
                 authToken: options.authToken,
-                url: endpoints[0].address,
+                url: options.url || endpoints[0].address,
```
Comment (ProbablePrime) [inline src/GameClient.ts]: We should probably avoid the external HTTP call here.

`if(options.url) { proimse.resolve().then(....)`

maybe?

**Comment by ProbablePrime** [comment]
@kateract specifically. Internal beam developers can run a copy of our interactive service locally. This would be a 122.0.0.1 address. This let's developers connect to those local copies.

File: src/GameClient.ts (hunk 2 above)
Comment (connor4312) [inline src/GameClient.ts]: Well, actually. Running the standalone service it exposes the a "fake" discovery endpoint which discovers itself. So alternately the option could be passed as an API base URL or something, which could be reused later if, for example, shortcode OAuth support is integrated into the library. (The standalone tetrisd service also exposes fake shortcode endpoints)

**Comment by kateract** [comment]
@ProbablePrime sorry, I didn't realize he was on your team, but the internal wording makes more sense now.

**Comment by ProbablePrime** [comment]
@saadataz I'll take over ownership of this and get it done, I'm more familiar with the requirements thanks!

**Comment by ProbablePrime** [comment]
I am now closing this as its goals are covered in #69

**Pull request closed without merging**